File: java/count_numbers_with_unique_digits.java
public class Solution {
    public int countNumbersWithUniqueDigits(int n) {
        if (n==0) return 1;
        if (n>10) return 0;
        int[] dp=new int[n];
        dp[0]=10;
        for (int i=1; i<n; i++) {
            if (i==1) dp[i]=9*9;
            else dp[i]=dp[i-1]*(10-i);
        }
        
        int res=0;
        for (int i=0; i<n; i++) res+=dp[i];
        return res;
    }
}

File: python/count-numbers-with-unique-digits.py
class Solution:
    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n==0: return 1
        n=min(n, 9)
        
        res, prod=10, 9
        for i in range(1, n):
            prod*=(10-i)
            res+=prod
            
        return res
        
